modules/storage/cosmos-db/module/cosmos.bicep

```diff
@@ -0,0 +1,286 @@
+param name _name
+param location _location
+param kind _kind = 'GlobalDocumentDB'
+param properties _properties
+
+resource cosmosDB 'Microsoft.DocumentDB/databaseAccounts@2022-05-15' = {
+  name: name
+  location: location
+  kind: kind
+  properties: properties
+}
+
+output id string = cosmosDB.id
+output name _name = cosmosDB.name
+
+type CosmosDB = {
+  name: _name
+  location: _location
+  tags: _tags?
+  kind: _kind?
+  identity: _identity?
+  properties: _properties
+}
+
+@description('The resource name.')
+@minLength(3)
+@maxLength(64)
+type _name = string
+
+@description('The location of the resource group to which the resource belongs.')
+type _location = 'eastus' | 'eastus2' | 'westus' | 'westus2' | 'westcentralus' | 'westeurope' | 'southeastasia' | 'northeurope' | 'uksouth' | 'ukwest' | 'australiaeast' | 'australiasoutheast' | 'brazilsouth' | 'southindia' | 'centralindia' | 'westindia' | 'canadacentral' | 'canadaeast' | 'japaneast' | 'japanwest' | 'koreacentral' | 'koreasouth' | 'francecentral' | 'southafricanorth' | 'uaenorth' | 'switzerlandnorth' | 'germanywestcentral' | 'norwayeast'
+
+@maxLength(128)
+type _tag_key = string
+
+@maxLength(256)
+type _tag_value = string
+
+@description('Tags are a list of key-value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key no greater than 128 characters and value no greater than 256 characters. For example, the default experience for a template type is set with "defaultExperience": "Cassandra". Current "defaultExperience" values also include "Table", "Graph", "DocumentDB", and "MongoDB".')
+@maxLength(15)
+type _tags = {
```

Comment:
I think we should set up a naming convention for types (and document that). Personally, I don't like mixing up camel case and snake case. How about we follow TypeScript's convention to use pascal case for types?

Reply:
sounds good. Let me add notes into the contribution guide with this PR to include these conventions.